### ISSUE TITLE
Show nothing in browser

### ISSUE THREAD

**Issue by trueMiskin** [TheOnlineJudge/ojudge#110]
I want try this out in docker, after I install ojudge, run it and everything looks good. But after access homepage, browser show me nothing.
Log of the first run: [output.log](https://github.com/TheOnlineJudge/ojudge/files/4538706/output.log)

There is my settings:
docker-compose.yml
```yml
version: '3'

services:
  ojudge:
    build: .
    command: build/bin/ojudgeapp --docroot "docroot;/favicon.ico,/css,/resources,/images" --resources-dir /usr/local/share/Wt/resources --http-address 127.0.0.1 --http-port 9090
    ports:
      - "9090:9090"
    depends_on:
      - db
  db:
    image: postgres
```
Dockerfile:
```Dockerfile
FROM debian:buster

RUN set -eux; \
    apt-get update; \
    \
    apt-get install -y git cmake g++ postgresql-server-dev-11 \
      graphicsmagick build-essential libboost-all-dev \
      qtbase5-dev pkg-config liboath-dev; \
    \
    git clone https://github.com/emweb/wt.git; \
    cd wt; \
    mkdir build; cd build; \
    cmake ../; make; make install; \
    cd ../..; \
    \
    apt-get update; \
    apt-get install -y libgraphicsmagick++1-dev libgraphicsmagick1-dev; \
    \
    git clone https://github.com/TheOnlineJudge/ojudge.git; \
    cd ojudge; \
    \
    cmake .; \
    make; \
    \
    mkdir -p ~/.config/ojudge/
    
ADD config /root/.config/ojudge/
WORKDIR /ojudge
```

config file
```
[database]
host=db
dbname=postgres
user=postgres
password=
```

Can you help me where is problem?

**Comment by IonutCicio** [comment]
I think you should create a new database instead of using the default postgres database, (I made a new database, called it "ojudge", and assigned a new user to it with full privileges); When I started everything it gave me an error because I left the password field empty, that's why assigned a new user to it. (this could be a problem too)

**Comment by trueMiskin** [comment]
I didnt think is this a problem. I add another dockerfile for postgres:
```dockerfile
FROM postgres
ENV POSTGRES_USER docker
ENV POSTGRES_PASSWORD mypass
ENV POSTGRES_DB ojudge
```
and update config (and update docker-compose):
```
[database]
host=db
dbname=ojudge
user=docker
password=mypass
```
But same problem, everything is ok, but browser cant show me anything ("Unable to connect"). Any error or message hasn't been shown during run.

**Comment by enjolras** [comment]
I see a couple of problems at least:

- libharu is not being installed, which is a required dependency
- cmake statement for Wt is missing some options, probably most important is: -DWT_WRASTERIMAGE_IMPLEMENTATION=GraphicsMagick

You have more info at https://groups.google.com/forum/#!topic/ojudge-developers/Tw9a2P16V0U, the 5th message explains how to build everything for mac, but it can be used as a guide to debian too (apart from the fact that most packages are provided by the distro).

My bet is that the RASTERIMAGE stuff is causing it to fail. I'm trying to build another docker image, based on archlinux instead of debian just to see if it would work. Please, try these changes and report back.

Thanks!

**Comment by trueMiskin** [comment]
I have been added libharu packege and options to dockerfile:
```dockerfile
FROM debian:buster

RUN set -eux; \
    apt-get update; \
    \
    apt-get install -y git cmake g++ postgresql-server-dev-11 \
      libgraphicsmagick++1-dev libgraphicsmagick1-dev \
      build-essential libboost-all-dev \
      libhpdf-2.3.0 libhpdf-dev graphicsmagick \
      qtbase5-dev pkg-config liboath-dev;

RUN set -eux; \
    git clone https://github.com/emweb/wt.git; \
    cd wt; \
    mkdir build; cd build; \
    cmake ../ -DWT_WRASTERIMAGE_IMPLEMENTATION=GraphicsMagick; make; make install; \
    cd ../..;

RUN set -eux; \
    git clone https://github.com/TheOnlineJudge/ojudge.git; \
    cd ojudge; \
    \
    cmake .; \
    make; \
    \
    mkdir -p ~/.config/ojudge/
    
ADD config /root/.config/ojudge/
WORKDIR /ojudge
```
But still when I go in browser to localhost:9090, it show me nothing. No error or anything what goes wrong.
I add logs from builds:
[webtoolkit-build.log](https://github.com/TheOnlineJudge/ojudge/files/4546072/webtoolkit-build.log)
[ojudge-build.log](https://github.com/TheOnlineJudge/ojudge/files/4546074/ojudge-build.log)

**Comment by trueMiskin** [comment]
Update: It works! I run docker in VM because on WIN home can not run it directly. So probably problem was in bad forwarded ports in VM.
My set up.
| Host IP   | Host Port | Guest IP | Guest Port | 
|-----------|-----------|----------|------------|
| 127.0.0.1 | 9090      |          | 9090       | 

It fix it up setting http address to 0.0.0.0.
I will quickly check if the first dockerfile fails to access the web.

**Comment by trueMiskin** [comment]
It seems that the web work with the first dockerfile, what I posted but if this project use wt graphics things then it will fail (it was not compiled). It should be fix by second dockerfile.

Running postgres from image works too. It isn't need have another user or database.

Some comments for compilation (what is good set it up)?
